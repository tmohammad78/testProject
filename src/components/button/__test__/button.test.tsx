import { cleanup, fireEvent, render } from '@testing-library/react'
import React from 'react'

import renderer from 'react-test-renderer';
import Button from '../index'

describe("Button component", () => {
    test("snapp shot test and check children value", () => {
        const component = renderer.create(<Button>Test</Button>)
        expect(component.toJSON()).toMatchSnapshot()
        expect(component.root.findByType("button").props.children).toEqual("Test");
    })
    test("handler click function as props should get data when clicked", (done) => {
        function handler(e) {
            expect(e.target.id).toEqual("1")
            done()
        }
        const { getByText } = render(<Button onClick={handler} >click me</Button>)
        const node = getByText("click me")
        fireEvent.click(node, { target: { id: 1 } })
    })
    afterEach(cleanup)
})