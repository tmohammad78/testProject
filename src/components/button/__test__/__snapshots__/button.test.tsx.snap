// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapp shot test and check children value 1`] = `
<button
  className="sc-bdfBwQ bHfbUj"
>
  Test
</button>
`;
